ivr/src/main/java/org/motechproject/ivr/service/impl/OutboundCallServiceImpl.java

```diff
@@ -130,6 +151,32 @@ public void initiateCall(String configName, Map<String, String> parameters) {
             addCallDetailRecord(CallDetailRecord.CALL_FAILED, config, params, motechCallId);
             throw new CallInitiationException(message, e);
         }
+    }
+
+    private void generateCallFile(Config config, Map<String, String> params, Map<String, String> completeParams, String motechCallId) throws URISyntaxException, IOException {
+        String myUri = mergeUriAndRemoveParams(config.getOutgoingCallUriTemplate(), completeParams);
+
+        Path path = Paths.get(new URI(myUri));
+        Path tempDirectory = Files.createTempDirectory(path.getParent(), "motech");
```

Comment:
Ok, but what if /tmp directory and asterisk call directory are on different filesystems? I don't know if this is a real problem, but this is possible. Problem with different filesystem is that could cause an exception during "atomic" move. I could not use atomic move and create file in temp directory or assume that existence of different filesystems on server is almost impossible.